doc/releases/changelog-dev.md

```diff
@@ -96,6 +98,7 @@
 
 <h4>Make use of more methods to map from molecules 🗺️</h4>
 
+
```

Comment:
@isaacdevlugt please can you fill in this section, corresponding to story 61894.

Please make a PR branching off of this one (`sc-61885-the-release-notes-are-polished`) and we can merge into here before finalizing everything.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/5575